code/mission/missionparse.cpp

```diff
@@ -5785,6 +5785,84 @@ void parse_variables()
 	}
 }
 
+void parse_sexp_containers()
+{
+	if (!optional_string("#Sexp_containers")) {
+		return;
+	}
+
+	if (optional_string("$Lists")) {
+		stuff_sexp_list_containers();
+		required_string("$End Lists");
+	}
+
+	if (optional_string("$Maps")) {
+		stuff_sexp_map_containers();
+		required_string("$End Maps");
+	}
+
+	// jg18 - persistence-related checking
+	// adapted from parse_variables()
+
+	// do this stuff only when playing through a campaign
+	if (Fred_running || !(Game_mode & GM_CAMPAIGN_MODE)) {
+		return;
+	}
+
+	// first update this mission's containers from campaign-persistent containers
+	for (const auto &current_pc : Campaign.persistent_containers) {
+		int index = get_sexp_container_index(current_pc.container_name.c_str());
+		if (index != -1) {
+			Assert(index >= 0 && index < (int)Sexp_containers.size());
+			auto &container = Sexp_containers[index];
+			// FIXME TODO: whatto do if theres a type mismatch between the two containers?
```

Comment:
If there's a mismatch I think we definitely have to use the one in the mission file rather than the persistent one since eternal containers could come from a completely unrelated campaign.

The game should give a warning but nothing beyond that.